### PR TITLE
Update Article.php

### PR DESCRIPTION
**Pull request by danilloricardo**
When article don't have tags, i believe that shoud be create defaults title_variant and image_variant.

### CODE REVIEW

File: Beam/app/Article.php [hunk 1]
```diff
@@ -104,13 +108,17 @@ public function getVariantsCountAttribute(): array
             if (!$result->count) {
                 continue;
             }
-            $titleVariant = $result->tags->title_variant;
-            $imageVariant = $result->tags->image_variant;
-
-            if (!array_key_exists($titleVariant, $variants)) {
-                $variants[$titleVariant] = [];
+            if ($result->tags) {
```
Comment (rootpd) [inline Beam/app/Article.php]: This condition is redundant, as tags are meant to be present there if `$result->count` has non-zero value.

File: Beam/app/Article.php [hunk 2]
```diff
@@ -104,13 +108,17 @@ public function getVariantsCountAttribute(): array
...
+                $titleVariant = $result->tags->title_variant;
+                $imageVariant = $result->tags->image_variant;
+                if (!array_key_exists($titleVariant, $variants)) {
+                    $variants[$titleVariant] = [];
+                }
+                $variants[$titleVariant][$imageVariant] = $result->count;
+            } else {
+                $variants[DEFAULT_TITLE_VARIANT][DEFAULT_IMAGE_VARIANT] = $result->count;
```
Comment (rootpd) [inline Beam/app/Article.php]: I don't think that defaults have any meaning here. If there are no values, there was no A/B test and we don't need to provide "single-variant" data for frontend. Single variant data are already displayed by frontend via the pageviews chart - the one with different color for each traffic source type.

**Comment by brunodelgado1988** [comment]
I'm watching this issue with Eric and Danillo, and maybe I can try to clarify what is happening.

What motivated @danilloricardo to open this issue is an exception that is thrown when we try to access the article detail page as you can see on the following image.

Screenshot from 2019-08-26 13-58-59

In fact, checking `$result->count` should work to prevent this error when article does not have tags. But here lies the problem.

When we are loading the Articles into Beam, the endpoint of articles upsert does not describes the Tags of the article, as you can see in [this payload body documentation] (https://github.com/remp2020/remp/tree/master/Beam#body).

The problem is that the Tracker Body describes the Tags, and it loads the information inside the Elastic Search. So, for Beam data saved on MySQL, the Article does not contains any tags. But for Elastic Search, it contains tags. So `$result->count` contains values greater than 0, but the object `$result->tags` contains Null value.

It's an problem caused by inconsistent data from MySQL and Elastic Search. MySQL does not contains the Tags from the Article, and Elastic Search contains because of the Tracker.

In the following image, you can see a var_dump() of  `$result->count` and `$result->tags` in the bottom of the image, under Conversion rate. While we have a count, we only have a Null inside tags.

Screenshot from 2019-08-26 14-14-09

File: Beam/app/Article.php (hunk 1 above)
Comment (rootpd) [reply Beam/app/Article.php]: This was a mistake on my side. The tags don't need to be there - it was only our CMS tracking the variant titles even for non-AB-tested articles.

File: Beam/app/Article.php (hunk 2 above)
Comment (rootpd) [reply Beam/app/Article.php]: Based on what was written above, the defaults now make sense.

**Pull request merged**